src/Symfony/Component/Semaphore/SemaphoreInterface.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Semaphore;
+
+use Symfony\Component\Semaphore\Exception\SemaphoreAcquiringException;
+use Symfony\Component\Semaphore\Exception\SemaphoreReleasingException;
+
+/**
+ * SemaphoreInterface defines an interface to manipulate the status of a semaphore.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+interface SemaphoreInterface
+{
+    /**
+     * Acquires the semaphore. If the semaphore has reached its limit.
+     *
+     * @return bool whether or not the semaphore had been acquired
+     *
+     * @throws SemaphoreAcquiringException If the semaphore can not be acquired
+     */
+    public function acquire(): bool;
```

Comment:
Why not allowing passing a number to acquire ? Which allow to do "weighted" acquire / release

(Same for release)

```suggestion
    public function acquire(int $weight): bool;
```

Reply:
What about?
```suggestion
    public function acquire(int $weight = 1): bool;
```

Reply:
I fixed it but in another way. now the value is in the key directly. It makes more sens IMHO